build/nightly-build/job-template.yml

```diff
@@ -0,0 +1,116 @@
+
+    steps:
+    - ${{ if eq(parameters.pool.name, 'Hosted macOS High Sierra') }}:
+      - script: brew update && brew install https://raw.githubusercontent.com/Homebrew/homebrew-core/f5b1ac99a7fba27c19cee0bc4f036775c889b359/Formula/libomp.rb && brew install mono-libgdiplus gettext && brew link gettext --force && brew link libomp --force
+        displayName: Install build dependencies
+    - ${{ if eq(parameters.name, 'Ubuntu_x64_NetCoreApp21') }}:
+      - bash: echo "##vso[task.setvariable variable=LD_LIBRARY_PATH](nightlyBuildRunPath):LD_LIBRARY_PATH"
+        displayName: Set LD_LIBRARY_PATH for Ubuntu to locate Native shared library
+    - script: parameters.buildScript−(_configuration) -buildArch=${{ parameters.architecture }}
+      displayName: Builds
+    - script: (dotnetPath)restore(nightlyBuildProjPath)
+      displayName: Restore nightly build project
+    - script: (dotnetPath)list(nightlyBuildProjPath) package --source (nugetFeed)−−outdated>(versionFilePath)
+      displayName: List latest package versions
+    - script: (dotnetPath)run−−project(packageUpdaterProjPath)
+      displayName: Update package version
+    - script: (dotnetPath)msbuild−restore(nightlyBuildProjPath) /p:ReferenceTypeForTestFramework="Nuget" /p:Configuration=(configuration)/p:TargetArchitecture={{ parameters.architecture }}
+      displayName: Build Nightly Build Project with latest package versions
+    - script: parameters.buildScript−(_configuration) -runnightlybuildtests
+      displayName: Run Nightly Build Tests
+      condition: succeeded()
+    - task: PublishTestResults@2
+      displayName: Publish Test Results
+      condition: succeededOrFailed()
+      inputs:
+        testRunner: 'vSTest'
+        searchFolder: '$(System.DefaultWorkingDirectory)/bin'
+        testResultsFiles: '**/*.trx'
+        testRunTitle: Machinelearning_Tests_Missing superscript or subscript argument(_configuration)_$(Build.BuildNumber)
+        configuration: $(_configuration)
+        mergeTestResults: true
+    - task: CopyFiles@2
+      displayName: Stage build logs
+      condition: not(succeeded())
+      inputs:
+        sourceFolder: $(Build.SourcesDirectory)
+        contents: '?(msbuild.*|binclash.log|init-tools.log)'
+        targetFolder: $(Build.ArtifactStagingDirectory)
+    - task: CopyFiles@2
+      displayName: Stage test output
+      condition: not(succeeded())
+      inputs:
+        sourceFolder: $(Build.SourcesDirectory)/bin
+        contents: |
+          **/TestOutput/**/*
+          **/*.trx
+        targetFolder: $(Build.ArtifactStagingDirectory)
+    - task: PublishBuildArtifacts@1
+      displayName: Publish build and test logs
+      condition: not(succeeded())
+      inputs:
+        pathToPublish: $(Build.ArtifactStagingDirectory)
+        artifactName: parameters.name(_config_short)
+        artifactType: container
+    - ${{ if eq(parameters.buildScript, 'build.cmd') }}:
+      - powershell: |
```

Comment:
This step to clean-up test data is not needed if you're not running any steps after running tests... the reason why we have it on CI builds is because we build packages after running tests and we need to cleanup space disk before we can do that to avoid failing because the machine ran out of space.

Reply:
I see, thanks for explain, will remove

---
In reply to: [343775233](https://github.com/dotnet/machinelearning/pull/4444#discussion_r343775233) [](ancestors = 343775233)